### PR TITLE
Feature/4877 server batch download

### PR DESCRIPTION
**Pull request by lukaselmer**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@
     scope '/:api_key' do
       scope '/content_blocks', controller: :content_blocks do
         post action: :store
+        get action: :batch_fetch
```
Comment (cyrilkyburz) [inline config/routes.rb]: is this only for batch download and a dashboard would habe another rout?

File: doc/renuo-cms-api/api/swagger/swagger.yaml [hunk 2]
```diff
@@ -90,7 +90,7 @@ paths:
           description: Unauthorized
```
Comment (cyrilkyburz) [inline doc/renuo-cms-api/api/swagger/swagger.yaml]: missing documentation for batch_fetch?

File: spec/requests/content_blocks_spec.rb [hunk 3]
```diff
@@ -4,17 +4,34 @@
   let!(:content_block) { create(:content_block) }
   let!(:credential_pair) { create(:credential_pair, api_key: content_block.api_key) }
 
+  # rubocop:disable Metrics/AbcSize
+  def compare_content_blocks(object, content_block)
```
Comment (cyrilkyburz) [inline spec/requests/content_blocks_spec.rb]: This kind of code is used often, would an own matcher help to keep it dry?

File: spec/controllers/content_blocks_spec.rb [hunk 4]
```diff
@@ -4,6 +4,35 @@
...
+    describe 'GET batch_fetch' do
+      # rubocop:disable Metrics/AbcSize
+      def check_object(blocks, index, content_block)
+        object = OpenStruct.new(blocks[index])
+        expect(object.content).to eq(content_block.content)
+        expect(object.content_path).to eq(content_block.content_path)
+        expect(object.api_key).to eq(content_block.api_key)
+        expect(object.id).to be_nil
+      end
+
```
Comment (cyrilkyburz) [inline spec/controllers/content_blocks_spec.rb]: blank line

File: spec/routing/content_blocks_routing_spec.rb [hunk 5]
```diff
@@ -3,12 +3,23 @@
 RSpec.describe 'routes for ContentBlocks', type: :routing do
   let(:api_base_path) { '/v1/sOmE4piK1/content_blocks' }
 
+  let(:basic_expected) do
+    { controller: 'v1/content_blocks', api_key: 'sOmE4piK1', format: 'json' }
```
Comment (cyrilkyburz) [inline spec/routing/content_blocks_routing_spec.rb]: nice

File: config/routes.rb (hunk 1 above)
Comment (lukaselmer) [inline config/routes.rb]: yes, it's the api part only. the dashboard may use this method - it's only the api part :wink:

File: doc/renuo-cms-api/api/swagger/swagger.yaml (hunk 2 above)
Comment (lukaselmer) [inline doc/renuo-cms-api/api/swagger/swagger.yaml]: it's right there, see line 93

Comment (lukaselmer) [inline doc/renuo-cms-api/api/swagger/swagger.yaml]: hmm, should the batch_fetch method be called index?

File: spec/controllers/content_blocks_spec.rb (hunk 4 above)
Comment (lukaselmer) [inline spec/controllers/content_blocks_spec.rb]: yep - autoformatted. i don't like it either... maybe i should simply not enable it anymore for this file?

File: spec/requests/content_blocks_spec.rb (hunk 3 above)
Comment (lukaselmer) [inline spec/requests/content_blocks_spec.rb]: hmm, good point. i think i only need it in this file though, so an own matcher would be an overkill?

File: spec/routing/content_blocks_routing_spec.rb (hunk 5 above)
Comment (lukaselmer) [inline spec/routing/content_blocks_routing_spec.rb]: :smile:

**Comment by lukaselmer** [comment]
@cyrilkyburz please review again :) thx

File: spec/routing/content_blocks_routing_spec.rb [hunk 6]
```diff
@@ -3,12 +3,23 @@
...
+
   it 'routes the create requests via POST' do
-    expect(post: "#{api_base_path}").to be_routable
+    expected = basic_expected.merge(action: 'store')
+    expect(post: "#{api_base_path}").to route_to(expected)
   end
 
   it 'routes the get requests via GET' do
-    expect(get: "#{api_base_path}/sample/path").to be_routable
+    expected = basic_expected.merge(action: 'fetch', content_path: 'sample/path')
```
Comment (cyrilkyburz) [inline spec/routing/content_blocks_routing_spec.rb]: should be index right?

**Comment by cyrilkyburz** [comment]
lgtm

**Comment by lukaselmer** [comment]
@cyrilkyburz thx :+1:

**Pull request merged**